### PR TITLE
Added missing Nullable annotations

### PR DESCRIPTION
**Pull request by rozza**
https://evergreen.mongodb.com/version/5ece156c0ae606788508de48

JAVA-3719

### CODE REVIEW

File: driver-core/src/main/com/mongodb/internal/connection/AbstractMultiServerCluster.java [hunk 1]
```diff
@@ -316,8 +318,9 @@ private boolean isStalePrimary(final ServerDescription newDescription) {
             return false;
         }
 
-        return (maxSetVersion.compareTo(newDescription.getSetVersion()) > 0
-                || (maxSetVersion.equals(newDescription.getSetVersion()) && maxElectionId.compareTo(newDescription.getElectionId()) > 0));
+        Integer setVersion = newDescription.getSetVersion();
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/connection/AbstractMultiServerCluster.java:321]: Why do we have to make this change for setVersion but not for electionId (used in `compareTo` below)?

File: driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java [hunk 2]
```diff
@@ -393,8 +393,13 @@ private MongoIncompatibleDriverException createIncompatibleException(final Clust
                     MIN_DRIVER_WIRE_VERSION, MIN_DRIVER_SERVER_VERSION);
         } else {
             incompatibleServer = curDescription.findServerIncompatiblyNewerThanDriver();
-            message = format("Server at %s requires wire version %d, but this version of the driver only supports up to %d.",
-                    incompatibleServer.getAddress(), incompatibleServer.getMinWireVersion(), MAX_DRIVER_WIRE_VERSION);
+            if (incompatibleServer != null) {
+                message = format("Server at %s requires wire version %d, but this version of the driver only supports up to %d.",
+                        incompatibleServer.getAddress(), incompatibleServer.getMinWireVersion(), MAX_DRIVER_WIRE_VERSION);
+            } else {
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java:399]: This conditional will never be entered.  Can we remove it and somehow disable the warning?

Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java:399]: We could throw an IllegalStateException exception.  Its all juggling for spotbugs that doesn't know if the method call has side effects or not.

Comment (jyemin) [reply driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java:399]: That seems better to me.

File: driver-core/src/main/com/mongodb/internal/connection/AbstractMultiServerCluster.java (hunk 1 above)
Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/connection/AbstractMultiServerCluster.java:321]: Because compareTo can take a null value?

Comment (jyemin) [reply driver-core/src/main/com/mongodb/internal/connection/AbstractMultiServerCluster.java:321]: Weird.  In IntellIJ I see `int compareTo(@NotNull T o)` and `public boolean equals(Object obj)`.  

Anyway, not that important so I'll resolve.

Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/connection/AbstractMultiServerCluster.java:321]: Ah I reverted it and then saw it - maxElectionId is an ObjectId so it's compareTo can handle null.

**Pull request closed without merging**